PhysicsTools/PatAlgos/python/slimming/applySubstructure_cff.py

```diff
@@ -12,105 +12,116 @@ def applySubstructure( process ) :
-    task.add(process.ak8PFJetsCHSTrimmedMass)
-    task.add(process.ak8PFJetsCHSFilteredMass)
-    task.add(process.ak8PFJetsCHSSoftDropMass)
-    process.patJetsAK8.userData.userFloats.src += ['ak8PFJetsCHSPrunedMass','ak8PFJetsCHSSoftDropMass']  
-    process.patJetsAK8.addTagInfos = cms.bool(False)
+    addToProcessAndTask('ak8PFJetsCHSPruned'+postfix, ak8PFJetsCHSPruned.clone(), process, task)
+    addToProcessAndTask('ak8PFJetsCHSSoftDrop'+postfix, ak8PFJetsCHSSoftDrop.clone(), process, task)
+    #process.load("RecoJets.JetProducers.ak8PFJetsCHS_groomingValueMaps_cfi")
+    #task.add(process.ak8PFJetsCHSPrunedMass)
+    #task.add(process.ak8PFJetsCHSTrimmedMass)
+    #task.add(process.ak8PFJetsCHSFilteredMass)
+    #task.add(process.ak8PFJetsCHSSoftDropMass)
+    from RecoJets.JetProducers.ak8PFJetsCHS_groomingValueMaps_cfi import ak8PFJetsCHSPrunedMass, ak8PFJetsCHSTrimmedMass, ak8PFJetsCHSFilteredMass, ak8PFJetsCHSSoftDropMass
+    addToProcessAndTask('ak8PFJetsCHSPrunedMass'+postfix, ak8PFJetsCHSPrunedMass.clone(), process, task)
+    addToProcessAndTask('ak8PFJetsCHSTrimmedMass'+postfix, ak8PFJetsCHSTrimmedMass.clone(), process, task)
+    addToProcessAndTask('ak8PFJetsCHSFilteredMass'+postfix, ak8PFJetsCHSFilteredMass.clone(), process, task)
+    addToProcessAndTask('ak8PFJetsCHSSoftDropMass'+postfix, ak8PFJetsCHSSoftDropMass.clone(), process, task)
+
+    getattr(process,"patJetsAK8").userData.userFloats.src += ['ak8PFJetsCHSPrunedMass'+postfix,'ak8PFJetsCHSSoftDropMass'+postfix]  
+    getattr(process,"patJetsAK8").addTagInfos = cms.bool(False)
 
 
 
     # add Njetiness
     process.load('RecoJets.JetProducers.nJettinessAdder_cfi')
     task.add(process.Njettiness)
-    addToProcessAndTask('NjettinessAK8', process.Njettiness.clone(), process, task)
+    addToProcessAndTask('NjettinessAK8'+postfix, process.Njettiness.clone(), process, task)
 
 
-    process.NjettinessAK8.src = cms.InputTag("ak8PFJetsCHS")
-    process.NjettinessAK8.cone = cms.double(0.8)
-    process.patJetsAK8.userData.userFloats.src += ['NjettinessAK8:tau1','NjettinessAK8:tau2','NjettinessAK8:tau3']
+    getattr(process,"NjettinessAK8").src = cms.InputTag("ak8PFJetsCHS"+postfix)
+    getattr(process,"NjettinessAK8").cone = cms.double(0.8)
+    getattr(process,"patJetsAK8").userData.userFloats.src += ['NjettinessAK8'+postfix+':tau1','NjettinessAK8'+postfix+':tau2','NjettinessAK8'+postfix+':tau3']
 
 
 
 
     #add AK8 from PUPPI
-    process.load('RecoJets.JetProducers.ak8PFJetsPuppi_cfi')
-    task.add(process.ak4PFJetsPuppi)
-    task.add(process.ak8PFJetsPuppi)
+    #process.load('RecoJets.JetProducers.ak8PFJetsPuppi_cfi')
+    from RecoJets.JetProducers.ak8PFJetsPuppi_cfi import ak4PFJetsPuppi, ak8PFJetsPuppi
+    addToProcessAndTask('ak4PFJetsPuppi'+postfix,ak4PFJetsPuppi.clone(), process, task)
+    addToProcessAndTask('ak8PFJetsPuppi'+postfix,ak8PFJetsPuppi.clone(), process, task)
+    #task.add(process.ak4PFJetsPuppi)
+    #task.add(process.ak8PFJetsPuppi)
```

Comment:
here and in a few more places in this file it looks like the new commented out lines are not needed (they are replaced by a more appropriate load of modules/tasks